src/mds/MDCache.cc

```diff
@@ -10625,13 +10620,9 @@ void MDCache::handle_discover_reply(const cref_t<MDiscoverReply> &m)
   } else if (m->is_flag_error_dn()) {
     frag_t fg = cur->pick_dirfrag(m->get_error_dentry());
     CDir *dir = cur->get_dirfrag(fg);
-    if (dir) {
-      if (dir->is_auth()) {
-	dir->take_sub_waiting(finished);
```

Comment:
How exactly is the migrator doing this already?

Reply:
dir->take_waiting(CDir::WAIT_ANY_MASK, waiters) In Migrator::decode_import_dir